machine-learning/sagemaker-edge-manager/artifacts/com.greengrass.SageMakerEdgeManager.ObjectDetection/1.0.0/object_detection/prediction_utils.py

```diff
@@ -0,0 +1,169 @@
+    UnLoadModelRequest,
+)
+
+config_utils.logger.info("Using np from '{}'.".format(np.__file__))
+config_utils.logger.info("Using cv2 from '{}'.".format(cv2.__file__))
+
+# Read labels file
+with open(config_utils.LABELS_FILE, "r") as f:
+    labels = literal_eval(f.read())
+
+
+def transform_image(im):
+    if len(im.shape) == 2:
+        height, width = im.shape[:2]
+        im = np.expand_dims(im, axis=2)
+        nchannels = 1
+    elif len(im.shape) == 3:
+        height, width, nchannels = im.shape[:3]
+    else:
+        raise Exception("Unknown image structure")
+    if nchannels == 1:
+        im = cv2.cvtColor(im, cv2.COLOR_GRAY2RGB)
+    elif nchannels == 4:
+        im = cv2.cvtColor(im, cv2.COLOR_BGRA2RGB)
+    elif nchannels == 3:
+        im = cv2.cvtColor(im, cv2.COLOR_BGR2RGB)
+    return im
+
+
+def predict_from_image(image):
+    r"""
+    Resize the image to the trained model input shape and predict using it.
+
+    :param image: numpy array of the image passed in for inference
+    """
+    img_data = transform_image(image)
+    img_data = cv2.resize(img_data, config_utils.SHAPE)
+    img_data = img_data[:, :, (2, 1, 0)].astype(np.float32)
+    img_data -= np.array([123, 117, 104])
+    img_data = np.transpose(np.array(img_data), (2, 0, 1))
+    img_data = np.expand_dims(img_data, axis=0)
+    predict(img_data)
+
+
+def load_image(image_path):
+    r"""
+    Validates the image type irrespective of its case. For eg. both .PNG and .png are valid image types.
+    Also, accepts numpy array images.
+
+    :param image_path: path of the image on the device.
+    :return: a numpy array of shape (1, input_shape_x, input_shape_y, no_of_channels)
+    """
+    # Case insenstive check of the image type.
+    img_lower = image_path.lower()
+    if (
+        img_lower.endswith(
+            ".jpg",
+            -4,
+        )
+        or img_lower.endswith(
+            ".png",
+            -4,
+        )
+        or img_lower.endswith(
+            ".jpeg",
+            -5,
+        )
+    ):
+        try:
+            image_data = cv2.imread(image_path)
+        except Exception as e:
+            config_utils.logger.error(
+                "Unable to read the image at: {}. Error: {}".format(image_path, e)
+            )
+            exit(1)
+    elif img_lower.endswith(
+        ".npy",
+        -4,
+    ):
+        image_data = np.load(image_path)
+    else:
+        config_utils.logger.error("Images of format jpg, jpeg, png and npy are only supported.")
+        exit(1)
+    return image_data
+
+
+def load_model(name, url):
+    load_request = LoadModelRequest()
+    load_request.url = url
+    load_request.name = name
+    config_utils.agent_client.LoadModel(load_request)
+
+
+def unload_model(name):
+    unload_request = UnLoadModelRequest()
+    unload_request.name = name
+    config_utils.agent_client.UnLoadModel(unload_request)
+
+
+def predict(image_data):
+    PAYLOAD = {}
+    PAYLOAD["timestamp"] = str(datetime.now())
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. The recommended way to create an aware datetime object representing a specific timestamp in UTC is by passing `tzinfo` as an argument to the method.

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)